src/pages/HistoryPage.js

```diff
@@ -29,58 +106,10 @@ const HistoryPage = props => {
       props.firebase
         .user(props.firebase.getMyUID())
         .get()
-        .then(user => {
-          Promise.all(
-            user.data().transactions.map(id =>
-              props.firebase
-                .transaction(id)
-                .get()
-                .then(transac => {
-                  transac = transac.data();
-                  transac.id = id;
-                  // calc transaction type
-                  if (transac.type === 'lend') {
-                    transac.type =
-                      transac.providerID === props.firebase.getMyUID()
-                        ? 'lend'
-                        : 'borrow';
-                  } else {
-                    transac.type =
-                      transac.providerID === props.firebase.getMyUID()
-                        ? 'give'
-                        : 'get';
-                  }
-
-                  return Promise.all([
-                    props.firebase
-                      // get the user other than me
-                      .user(
-                        transac.providerID === props.firebase.getMyUID()
-                          ? transac.consumerID
-                          : transac.providerID
-                      )
-                      .get()
-                      .then(
-                        // inject the other user's data
-                        user => (transac.involvedUser = user.data())
-                      ),
-                    props.firebase
-                      .book(transac.itemID)
-                      .get()
-                      .then(
-                        // inject book data
-                        book => (transac.book = book.data())
-                      )
-                  ]).then(bookAndUser => {
-                    return transac;
-                  });
-                })
-            )
-          ).then(transacsWithData => {
-            setTransactions(transacsWithData);
-            setGotTransactions(true);
-          });
-        });
+        .then(
+          user =>
+            user.data().transactions ? manageTransactions(user) : <Loader /> //noTransactionsToShow
```

Comment:
change to this:
user.date().transactions && user.data().transactions.length>0 ? ......